### PR TITLE
fix: update webgpu api and fix config

### LINKED ISSUE #2: workgroup_size(1, 1, 1) is inefficient
Using `workgroup_size(1, 1, 1)` means that out of the N (8/32/64 depending on HW) ALUs of your GPU's execution units only one is doing work. You'd likely get a multiple X improvement in runtime by using a bigger workgroup size. (that's why WGSL forces people to type `workgroup_size`). 64 is a good default for the volume (to be split over the dimensions as is convenient). Be careful though that only whole workgroups are scheduled so you might need to add checks for the `global_id` being in-bounds.

### LINKED ISSUE #5: live demo isn't working due to webgpu api has changed
please review this [pull request](https://github.com/harp-lab/GraphWaGu/pull/4) if you have time

### PR DESCRIPTION
**Pull request by quarksb**
Closes #5 

- dispatch() is replace with dispatchWorkgroups()
- pnpm is much faster than npm, you will enjoy it

### CODE REVIEW

**Comment by ldyken53** [comment]
What browser are you using? The demo still works in Chrome and Edge as of now, I believe the API changes you fix with this pull request have not been pushed to these browsers yet. Those are the main targets with this demo, so I plan on waiting until the change takes place for those browsers before merging this request.

**Comment by pravinpoudel** [comment]
I appreciate the changes and update you made.

They have been doing a lot of changes in WebGPU and WGSL in the last couple of months so before doing any update it should be assured these are stable changes. 

Just for head up, the demo is working fine in the Edge browser but yes, not in Canary. Did you check on Edge?

**Comment by quarksb** [comment]
sorry, I only check it in Canary……, you are right for "waiting until the change takes place for those browsers before merging this request."
[my demo for canary](https://quarksb.github.io/GraphWaGu/)

File: src/components/Sidebar.tsx [hunk 1]
```diff
@@ -4,13 +4,13 @@ import Collapsible from 'react-collapsible';
 import Select from 'react-select';
 function importAll(r) {
     let datasets = {};
-    r.keys().map((item, index) => { datasets[item.replace('./', '').replace('.png', '')] = r(item); });
+    r.keys().forEach((item) => { datasets[item.replace('./', '').replace('.png', '')] = r(item); });
```
Comment (quarksb) [inline src/components/Sidebar.tsx]: if you return nothing(or void), Array.forEach is better than Array.map

File: src/webgpu/force_directed.ts [hunk 2]
```diff
@@ -69,6 +67,7 @@ class ForceDirected {
         });
 
         this.createMatrixPipeline = device.createComputePipeline({
+            layout: 'auto',
```
Comment (quarksb) [inline src/webgpu/force_directed.ts:70]: layout is need for now

File: src/webgpu/render.tsx [hunk 3]
```diff
@@ -45,7 +45,7 @@ class Renderer {
       canvasRef.current.clientWidth * devicePixelRatio,
       canvasRef.current.clientHeight * devicePixelRatio,
     ];
-    const presentationFormat = context.getPreferredFormat(adapter);
```
Comment (quarksb) [inline src/webgpu/render.tsx:48]: getPreferredFormat is deprecated

**Pull request merged**